ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -347,6 +347,37 @@ let dss_loadavg () =
     )
   ]
 
+let dss_hostload xc domains =
+  let physinfo = Xenctrl.physinfo xc in
+  let pcpus = physinfo.Xenctrl.nr_cpus in
+  let rec loop n f = if n < 0 then 0 else f n + loop (n - 1) f in
+  let load =
+    List.fold_left
+      (fun acc (dom, _, domid) ->
+        loop dom.Xenctrl.max_vcpu_id (fun id ->
+            let vcpuinfo = Xenctrl.domain_get_vcpuinfo xc domid id in
+            if vcpuinfo.Xenctrl.online && not vcpuinfo.Xenctrl.blocked then
+              1
+            else
+              0
+        )
+        + acc
+      )
+      0 domains
+  in
+  let load_per_cpu = float_of_int load /. float_of_int pcpus in
+  [
+    ( Rrd.Host
+    , Ds.ds_make ~name:"hostload" ~units:"(fraction)"
+        ~description:
+          ("Host load per physical cpu, where load refers to "
+          ^ "the number of vGPU(s) in running or runnable status."
```

Comment:
How about `the ratio between the number of running/runnable vCPUs and the number of physical CPU cores`

Reply:
There exists another ds `loadavg` which `load` is defined in "uptime" and "top", we just extend the concept to host.